src/core/eval.ml

```diff
@@ -77,38 +79,55 @@ let snf : (term -> term) -> (term -> term) = fun whnf ->
     | TRef(_)     -> assert false
   in snf
 
-(** Configuration of the reduction engine. *)
-type config =
-  { context : ctxt (** Context of the reduction used for generating metas. *)
-  ; defmap : term VarMap.t (** Variable definitions. *)
-  ; rewrite : bool (** Use user-defined rewrite rules. *)
-  ; problem : problem (** Generated metavariables. *) }
-
-(*let pp_defmap = D.map VarMap.iter pp_var " ≔ " pp_term "; "*)
+type rw_tag = [ `NoBeta | `NoRw | `NoExpand ]
```

Comment:
Why negative tags instead of positive tags? I would prefer positive tags (Beta, Rewrite, Def).

Reply:
Why using backquotes?

Reply:
I used negative tags because I assume that the default behaviour is to have the full rewrite engine. So default would be `[Beta; Rw; Expand]`, but we don't want to pass all these arguments each time. Backquotes allow to use again these constructors for other types, it's mainly for convenience (we can use ordinary variants).

Reply:
Using backquotes also prevents from the need to qualify the identifiers, so in `unif.ml` we can write ``Eval.whnf [`NoRw]`` instead of ``Eval.whnf [Eval.NoRw]`` (and instead of ``Eval.whnf [Eval.Beta; Eval.Expand]`` if we take positive tags). It's all for convenience.